graph/src/components/store/traits.rs

```diff
@@ -474,14 +474,16 @@ pub trait ChainStore: Send + Sync + 'static {
     ) -> Result<Vec<serde_json::Value>, Error>;
 
     /// Get the `offset`th ancestor of `block_hash`, where offset=0 means the block matching
-    /// `block_hash` and offset=1 means its parent. Returns None if unable to complete due to
-    /// missing blocks in the chain store.
+    /// `block_hash` and offset=1 means its parent. If `root` is passed, short-circuit upon finding
+    /// a child of `root`. Returns None if unable to complete due to missing blocks in the chain
+    /// store.
```

Comment:
I don't understand what 'short-circuiting' here means. Does that mean that when `root` is passed that the search stops if we either get to the offset-th ancestor of `block_ptr` or find a block whose parent is `root`?

Reply:
Yes that is right

Reply:
I looked at it again, and I think I misread the changes the first time around. You are right, it's not adding an RPC call, just changed how a call is handled. So looks good to me.

One minor nit is that the method name `nearest_block_hash_to_number` sounds like it's translating a hash to a number when it is doing the reverse, maybe calling it `nearest_block_hash_for_number` would be clearer.